client/app/scripts/utils/web-api-utils.js

```diff
@@ -157,3 +176,42 @@ export function doControl(probeId, nodeId, control) {
     }
   });
 }
+
+export function deletePipe(pipeId) {
+  const url = `api/pipe/${encodeURIComponent(pipeId)}`;
+  reqwest({
+    method: 'DELETE',
+    url: url,
+    success: function() {
+      log('Closed the pipe!');
+    },
+    error: function(err) {
+      log('Error closing pipe:' + err);
+      receiveError(url);
+    }
+  });
+}
+
+export function getPipeStatus(pipeId) {
+  const url = `/api/pipe/${encodeURIComponent(pipeId)}`;
+  reqwest({
+    method: 'GET',
+    url: url,
+    success: function(res) {
+      log('ERROR: expected responses: [400, 404]. Got:', res);
+    },
+    error: function(err) {
+      const status = {
+        400: 'PIPE_ALIVE',
```

Comment:
True! Yeah this is a bit of a hack :(.

1. Websockets don't receive the http status code from the server (for security reasons). Thats why we do this request.
2. If a pipe exists the server expects a WS connection, but we're xhr'ing which it considers a bad request (400).
3. If a pipe doesn't exist it returns a 404 and doesn't care that we're xhr'ing.

So it would be nice if either:
 - the server could respond w/ a 200 if we're xhr'ing and a WS handshake if we're WS'ing, (I'm not sure if thats easy).
 - We had another endpoint that was just for xhr GETting etc. e.g. `/api/pipe/:pid/status`.

We could open a new ticket for that, as this might causes scares in the future.